ref/Microsoft.Build.Framework/net/Microsoft.Build.Framework.cs

```diff
@@ -242,6 +242,22 @@ public partial interface IEventSource3 : Microsoft.Build.Framework.IEventSource,
         void IncludeEvaluationProfiles();
         void IncludeTaskInputs();
     }
+    public partial interface IFileSystem
+    {
+        void ClearCaches();
+        bool DirectoryEntryExists(string path);
```

Comment:
What's a "directory entry" vs just a directory?

Reply:
An entry is either a file or a directory. Took the naming from [this BCL method](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.enumeratefilesystementries?view=netcore-3.1). Is `FileOrDirectoryExists` better? Longer but explicit.